frontend/src/scenes/data-warehouse/new/sourceWizardLogic.tsx

```diff
@@ -433,5 +441,105 @@ export const sourceWizardLogic = kea<sourceWizardLogicType>([
                 actions.setIsLoading(false)
             }
         },
+        handleRedirect: async ({ kind, searchParams }) => {
+            switch (kind) {
+                case 'hubspot': {
+                    actions.updateSource({
+                        source_type: 'Hubspot',
+                        payload: {
+                            code: searchParams.code,
+                            redirect_uri: getHubspotRedirectUri(),
+                        },
+                    })
+                    return
+                }
+                default:
+                    lemonToast.error(`Something went wrong.`)
+            }
+        },
+        submitSourceConnectionDetailsSuccess: () => {
+            actions.getDatabaseSchemas()
+        },
+        getDatabaseSchemas: async () => {
+            if (values.selectedConnector) {
+                const schemas = await api.externalDataSources.database_schema(
+                    values.selectedConnector.name,
+                    values.source.payload ?? {}
+                )
+                actions.setDatabaseSchemas(schemas)
+                actions.onNext()
+            }
+        },
+    })),
+    urlToAction(({ actions }) => ({
+        '/data-warehouse/:kind/redirect': ({ kind = '' }, searchParams) => {
```

Comment:
two callbacks here so that there's a dedicated route that can be used and defined in the hubspot config/any other oauth config in the future